### PR TITLE
python: Typing files fixes.

### PR DESCRIPTION
**Pull request by da-tanabe**
python: Fixes to typing files.

dazl's mypy.ini file formerly _excluded_ `dazl._gen` for historical reasons, but since typings files were introduced for these files in #192 and #271, this exclusion should have been removed. But as it turns out, these exclusions were hiding the fact that there were some bugs in the typing rules, notably around Python 3.6/3.7 support. This PR removes those exclusions and fixes bugs in the typing files.

### CODE REVIEW

File: python/dazl/protocols/v1/pb_parse_event.py [hunk 1]
```diff
@@ -239,18 +239,14 @@ def serialize_transactions_request(
...
+        ledger_offset = lapipb.LedgerOffset(boundary=0)
 
     if f.destination_offset is not None:
-        final_offset = lapipb.LedgerOffset()
-        final_offset.absolute = f.destination_offset
+        final_offset = lapipb.LedgerOffset(absolute=f.destination_offset)
     else:
-        final_offset = lapipb.LedgerOffset()
-        final_offset.boundary = 1
+        final_offset = lapipb.LedgerOffset(boundary=1)
```
Comment (mschaef-da) [inline python/dazl/protocols/v1/pb_parse_event.py:249]: Typing issues aside, I like this better anyway.

File: python/dazl/protocols/v1/pb_parse_event.py [hunk 2]
```diff
@@ -432,11 +428,12 @@ def to_event(
     context: "Union[TransactionEventDeserializationContext, ActiveContractSetEventDeserializationContext]",
     evt_pb: "Union[lapipb.Event, lapipb.TreeEvent]",
 ) -> "Optional[OffsetEvent]":
+    # TODO: mypy should understand that
     try:
-        event_type = evt_pb.WhichOneof("event")
+        event_type = evt_pb.WhichOneof("event")  # type: ignore
```
Comment (mschaef-da) [inline python/dazl/protocols/v1/pb_parse_event.py:433]: Not so excited about this.

**Review by mschaef-da — APPROVED** [review]
LGTM

File: python/dazl/protocols/v1/pb_parse_event.py (hunk 1 above)
Comment (da-tanabe) [reply python/dazl/protocols/v1/pb_parse_event.py:249]: indeed

File: python/dazl/protocols/v1/pb_parse_event.py (hunk 2 above)
Comment (da-tanabe) [reply python/dazl/protocols/v1/pb_parse_event.py:433]: yeah I don't know why `mypy` has trouble with the `Union` in this context. Nevertheless, this whole file is pending deprecation anyway thank goodness.

**Pull request merged**